test/batch.js

```diff
@@ -23,17 +23,24 @@ test('create nodes', function (t) {
     }
   ]
 
+  var updatedWithLinks = nodes
+  updatedWithLinks[0].links = []
+  updatedWithLinks[1].links = []
```

Comment:
You are also updating `nodes` here. If you put `t.deepEquals(nodes, updatedWithLinks)` you will find they are equal (also, `t.equal(nodes, updatedWithLinks)`, since they both reference the same object. This means that the test on line 42 `t.deepEquals(elms, updatedWithLinks)` is not actually testing anything. If you want to check that the returned arguments `elms` of batch is updated with links, then you need to clone `nodes` to created an expected value.

Reply:
You're totally right @gmaclennan, thanks! Haven't had to deal with mutating objects in a while.